### DISCUSSION TITLE
can't config texcat

### DISCUSSION THREAD

**Discussion by dtaubaso** [explosion/spaCy#7742] [Help: Coding & Implementations] [answered]
Hi everyone, 

I'm using spaCy 3.0 on Win 10 (64 bit), trying to do a Jupyter Notebook using textcat.

I'm following several tutorials, but they all seem to have been written with Spacy 2 in mind.

I'm really new into this, so I'd really appreciate any help.

I'm trying to do apply textcat with the single_label_cnn_config.

I've imported from spacy.pipeline.textcat import single_label_cnn_config

Then, following @svlandeg suggestion on another question, made

```
model = { "@architectures": "spacy.TextCatCNN.v1", "exclusive_classes": True,}

config = {"model": model}
```

But it returns "textcat.model -> tok2vec field required"

Isn't there a way to just call "model", do I have to define "model.tok2vec" too?

How do I do that, how do I config texcat?

Thank you all

**Comment by polm** [comment, marked as answer]
When you specify a config you generally have to specify all options available to a model, so yes, you need to define the tok2vec component. One of the design goals with the spaCy configs is to make all settings explicit so that changes in library versions or environment details don't change how a config works later.

If you're getting started with textcat I'd recommend trying the [Quickstart](https://spacy.io/usage/training#quickstart) first to generate a working config, and then tweaking the settings after you know it works. You can also look at [these tutorials](https://github.com/explosion/projects/tree/v3/tutorials) for complete pipelines you can run and fiddle with.

**Reply by dtaubaso** [reply]
thank you!
